### ISSUE TITLE
[CRASH] Shutdown is not clean when closing the window while cache is occuring

### ISSUE THREAD

**Issue by aeroc7** [olive-editor/olive#1899]
**Commit Hash** 
3661599

**Platform** 
All

**Summary**
If a pre-cache is occurring and the window is closed, a seg fault will occur. This is presumably due to some part of the cache system not receiving a proper signal to shutdown, so while other parts of the program are being closed down, the cache system continues on by itself.

**Steps to Reproduce**

1. Import a clip & pre-cache it
2. Close the window (save or don't save, it doesn't matter)
3. Seg fault will occur

<details><summary><strong>Crash Report</strong></summary><pre><code>

</code></pre></details>

**Additional Information**

**Comment by ThomasWilshaw** [comment]
Are you able to provide a stack trace?

**Comment by aeroc7** [comment]
> Are you able to provide a stack trace?

Yes, updated. Just not sure how useful it is, as the stack trace actually seemed to differ from the several times I tried it, which is what led me to believe something wasn't getting shutdown in the proper order.
